src/app/modules/auth.rs

```diff
@@ -1,18 +1,246 @@
-use crate::app::store::ProvableStore;
-use crate::app::BaseCoinApp;
+use crate::app::modules::bank::Denom;
+use crate::app::modules::Module;
+use crate::app::store::{
+    Height, Path, ProtobufStore, ProvableStore, SharedStore, Store, TypedStore,
+};
+use crate::prostgen::cosmos::auth::v1beta1::BaseAccount;
 use crate::prostgen::cosmos::auth::v1beta1::{
-    query_server::Query as AuthQuery, QueryAccountRequest, QueryAccountResponse,
-    QueryAccountsRequest, QueryAccountsResponse, QueryParamsRequest as AuthQueryParamsRequest,
-    QueryParamsResponse as AuthQueryParamsResponse,
+    query_server::{Query, QueryServer},
+    QueryAccountRequest, QueryAccountResponse, QueryAccountsRequest, QueryAccountsResponse,
+    QueryParamsRequest, QueryParamsResponse,
 };
 
+use std::collections::HashMap;
+use std::convert::{TryFrom, TryInto};
+use std::str::FromStr;
+
+use cosmrs::AccountId;
 use prost::Message;
 use prost_types::Any;
+use serde_json::Value;
+use tendermint_proto::Protobuf;
 use tonic::{Request, Response, Status};
-use tracing::debug;
+use tracing::{debug, trace};
+
+const RELAYER_ACCOUNT: &str = "cosmos1snd5m4h0wt5ur55d47vpxla389r2xkf8dl6g9w";
```

Comment:
can we clarify what this account is?

Reply:
This is basically the account that hermes must configured with using the `hermes keys add basecoin-0` and that hermes uses to sign basecoin transactions. At the moment, this is hardcoded and works because basecoin doesn't verify signatures. I opened an issue to fix this -> https://github.com/informalsystems/basecoin-rs/issues/52, but for now we must certainly document this. 👍

Reply:
https://github.com/informalsystems/basecoin-rs/pull/44/commits/9aebdad07c92890838dffa58798b2f43f459d3f0